styx-api-service/src/main/java/com/spotify/styx/StyxApi.java

```diff
@@ -100,13 +102,16 @@ public void create(Environment environment) {
     final StyxConfigResource styxConfigResource = new StyxConfigResource(storage);
     final CliResource cliResource = new CliResource(schedulerServiceBaseUrl, storage);
 
+    final Middleware clientValidator = Middlewares.clientValidator(storage);
```

Comment:
I guess there's one difference in that `CachedSupplier` supports injecting time while the Guava decorator uses wall clock time.